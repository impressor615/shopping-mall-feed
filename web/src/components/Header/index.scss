@import "~@/assets/styles/variables";

header {
  box-sizing: border-box;
  position: fixed;
  top: 0;
  width: 100%;
  max-width: 1100px;

  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 10px 20px;

  background: $lightgrey;
  box-shadow: 0 3px 5px $secondary;
}