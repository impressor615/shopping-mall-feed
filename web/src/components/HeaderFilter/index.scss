@import "~@/assets/styles/variables";

.header-filter {
  display: inline-block;

  .link:first-child {
    border-right: none;
    border-radius: 10px 0 0 10px;
  }

  .link:last-child {
    border-radius: 0 10px 10px 0;
  }
}