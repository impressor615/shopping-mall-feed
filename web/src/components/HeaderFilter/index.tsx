import "./index.scss";

import React from "react";

import Button from "@/components/Button";

interface HeaderFilterProps {
  filter: string;
  onClick: (e: React.MouseEvent) => void;
}

const HeaderFilter = ({ filter, onClick }: HeaderFilterProps) => (
  <div className="header-filter" onClick={onClick}>
    <Button name="products" active={filter === "products"}>상품</Button>
    <Button name="ranks" active={filter === "ranks"}>랭킹</Button>
  </div>
);

export default HeaderFilter;
