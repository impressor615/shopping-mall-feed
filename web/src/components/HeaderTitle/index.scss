@import "~@/assets/styles/variables";

.header-title {
  span:first-child {
    margin-right: 15px;
  }

  span:last-child {
    font-size: 15px;
    color: $secondary;
  }
}