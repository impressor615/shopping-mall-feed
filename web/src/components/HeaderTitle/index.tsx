import "./index.scss";

import React from "react";

interface HeaderTitleProps {
  title: string;
}

export const getToday = (): string => {
  const currentYear = new Date().getFullYear().toString().substr(-2);
  const currentMonth = new Date().getMonth() + 1;
  const currentDate = new Date().getDate();
  return `${currentYear}. ${currentMonth}. ${currentDate}`;
};

const HeaderTitle = ({ title }: HeaderTitleProps) => (
  <h3 className="header-title">
    <span>{title}</span>
    <span>{getToday()}</span>
  </h3>
);

export default HeaderTitle;
