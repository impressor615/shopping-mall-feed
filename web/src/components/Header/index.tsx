import "./index.scss";

import React from "react";

import HeaderFilter from "@/components/HeaderFilter";
import HeaderTitle from "@/components/HeaderTitle";

interface HeaderProps {
  filter: string;
  onClick: () => void;
}

const Header = ({ filter, onClick }: HeaderProps) => (
  <header>
    <HeaderTitle title="오늘 뭐 입지?" />
    <HeaderFilter filter={filter} onClick={onClick} />
  </header>
);

export default Header;
