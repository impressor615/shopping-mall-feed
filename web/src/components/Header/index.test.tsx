import enzyme from "enzyme";
import React from "react";

import Header from "@/components/Header";
import HeaderFilter from "@/components/HeaderFilter";
import HeaderTitle from "@/components/HeaderTitle";

describe("Header", () => {
  const filter = "products";
  const onClick = jest.fn();
  it("should render HeaderTitle and HeaderFilter", () => {
    const wrapper = enzyme.shallow(<Header filter={filter} onClick={onClick} />);
    expect(wrapper.find(HeaderFilter)).toHaveLength(1);
    expect(wrapper.find(HeaderTitle)).toHaveLength(1);
  });

  it("should be matched with snapshot", () => {
    const wrapper = enzyme.shallow(<Header filter={filter} onClick={onClick} />);
    expect(wrapper).toMatchSnapshot();
  });
});
