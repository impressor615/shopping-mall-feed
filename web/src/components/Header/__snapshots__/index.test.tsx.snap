// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should be matched with snapshot 1`] = `
<header>
  <HeaderTitle
    title="오늘 뭐 입지?"
  />
  <HeaderFilter
    filter="products"
    onClick={[MockFunction]}
  />
</header>
`;
